streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -254,6 +254,32 @@ public void testCannotStartTwice() throws Exception {
         }
     }
 
+    @Test(expected = IllegalStateException.class)
```

Comment:
We should not expect an exception here but add a `fail(...)` after `streams.setUncaughtExceptionHandler(null);` -- and thus, not rethrow within catch.